packages/astro/src/runtime/server/render/common.ts

```diff
@@ -89,6 +89,13 @@ function stringifyChunk(
 				}
 				return renderAllHeadContent(result);
 			}
+			case 'renderer-hydration': {
+				if (!result._metadata.hasRendererSpecificHydrationScript[instruction.rendererName]) {
```

Comment:
I don't understand this check, why do we check against `false` values? Shouldn't we **remove** the `if` block?

Reply:
The [Solid hydration script](https://www.solidjs.com/guides/server#hydration-script) only needs to be rendered once per page, so this is a check to make sure it has not been rendered yet. If we removed the `if`, it would get rendered once per component instead, which would probably still work, but would not be ideal as it would be sending some redundant data to the client.